tex2wiki/src/tex2wiki.py

```diff
@@ -0,0 +1,413 @@
+
+            # print metadata
+
+            equations[i] = LatexEquation(formula, '\n'.join(equation), metadata)
+
+        result.append([section_data[0], equations])
+
+    return result
+
+
+def find_end(text, left_delimiter, right_delimiter, start=0):
+    net = 0  # left delimiters encountered - right delimiters encountered
+    for i, ch in enumerate(text[start:]):
+        if ch == left_delimiter:
+            net += 1
+        elif ch == right_delimiter:
+            if net == 1:
+                return i + start
+            else:
+                net -= 1
+
+    return -1
+
+
+def get_data_str(text, latex=""):
+    if latex + "{" not in text:
+        return ""
+
+    start = text.find(latex + "{")
+    return text[start + len(latex + "{"):find_end(text, "{", "}", start)]
+
+
+def generate_nav_bar(info):
+    links = list()
+    for link, text in info:
+        link = link.replace("''", "")
+        text = text.replace("''", "")
+        links.append(generate_link(link, text))
+
+    nav_section = generate_html("div", "<< " + links[0], options={"id": "alignleft"}, spacing=1) + "\n" + \
+                  generate_html("div", links[1], options={"id": "aligncenter"}, spacing=1) + "\n" + \
+                  generate_html("div", links[2] + " >>", options={"id": "alignright"}, spacing=1)
+
+    header = generate_html("div", nav_section, options={"id": "drmf_head"})
+    footer = generate_html("div", nav_section, options={"id": "drmf_foot"})
+
+    return header, footer
+
+
+def get_macro_name(macro):
+    # (str) -> str
+    """Obtains the macro name."""
+    macro_name = ""
+    for ch in macro:
+        if ch.isalpha() or ch == "\\":
+            macro_name += ch
+        elif ch in ["@", "{", "["]:
+            break
+
+    return macro_name
+
+
+def find_all(pattern, string):
+    # (str, str) -> generator
+    """Finds all instances of pattern in string."""
+
+    i = string.find(pattern)
+    while i != -1:
+        yield i
+        i = string.find(pattern, i + 1)
+
+
+def get_symbols(text, glossary):
+    # (str, dict) -> str
+    """Generates span text based on symbols present in text."""
+    symbols = set()
+
+    for keyword in glossary:
+        for index in find_all(keyword, text):
+            # if the macro is present in the text
+            if index != -1:
+                index += len(keyword)  # now index of next character
+                if index >= len(text) or not text[index].isalpha():
+                    symbols.add(keyword)
+
+    span_text = ""
+    for symbol in sorted(symbols, key=str.lower):
+        links = list()
+        for cell in glossary[symbol]:
+            if "http://" in cell or "https://" in cell:
+                links.append(cell)
+
+        id_link = links[0]
+        links = ["[" + link + " " + link + "]" for link in links]
+
+        meaning = list(glossary[symbol][1])
+
+        count = 0
+        for i, ch in enumerate(meaning):
```

Comment:
Was this not already done in the dollar signs function? If not, is it possible to adapt that to allow it to be used here to avoid code duplication?